### PR TITLE
adding in open api examples that follow our guidelines

### PR DESCRIPTION
**Pull request by ianqueue**
~~This is just a start... one thing I'm not sure on is how to link this to the swagger editor, but I will update as soon as I figure that out.~~

Using bookmarks as an example, feedback appreciated!

### CODE REVIEW

**Comment by esimons** [comment]
@ianqueue the `url` parameter does it, I believe:
https://editor.swagger.io/?url=https://raw.githubusercontent.com/EBSCOIS/api-standards/8e2f185e79d6bd961ceeaf6334c797330fb2751f/examples.json

File: examples.json [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    },
+    "Category": {
+      "type": "object",
+      "properties": {
+        "id": { "type": "integer" },
+        "name": { "type": "string" }
+      },
+      "xml": { "name": "Category" }
+    },
+    "Contributors": {
```
Comment (esimons) [inline examples.json]: This should be singular `Contributor`, I think.

Comment (esimons) [inline examples.json]: This should be singular `Contributor`, I think.
```suggestion
    "Contributor": {
```

File: examples.json [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+      "type": "object",
+      "properties": {
+        "an": {
+          "type": "string",
+          "example": "vmc.b30177686"
+        },
+        "contributors": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Contributors"
```
Comment (esimons) [inline examples.json]: This should be singular `Contributor`, I think.
```suggestion
            "$ref": "#/definitions/Contributor"
```

File: examples.json [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+    "PersonalizedHeader": {
+      "in": "header",
+      "name": "X-EIS-ID-TOKEN",
+      "description": "Service identity header.",
+      "type": "string",
+      "required": true,
+      "default": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxIiwibmFtZSI6IkEiLCJpYXQiOjEsImVic2NvX3B1YV9pZCI6IjA3MjQyZjlhIn0.nRe7fIjMAIf1SkpFOTAEga3Gxx_Tk_QUgwATrgZpGfg"
+    }
+  },
+  "securityDefinitions": {
```
Comment (esimons) [inline examples.json]: Not sure what this should be, but imagine this whole block's gotta change.

File: examples.json [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+    }
+  },
+  "parameters": {
+    "BearerHeader": {
+      "in": "header",
+      "name": "Authorization",
+      "description": "JWT token required to get resources.",
+      "type": "string",
+      "required": true,
+      "default": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxIiwibmFtZSI6IkEiLCJpYXQiOjEsImVic2NvX3B1YV9pZCI6IjA3MjQyZjlhIn0.nRe7fIjMAIf1SkpFOTAEga3Gxx_Tk_QUgwATrgZpGfg"
```
Comment (esimons) [inline examples.json]: No `default`, right?
```suggestion
```

File: examples.json [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+    },
+    "PersonalizedHeader": {
+      "in": "header",
+      "name": "X-EIS-ID-TOKEN",
+      "description": "Service identity header.",
+      "type": "string",
+      "required": true,
+      "default": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxIiwibmFtZSI6IkEiLCJpYXQiOjEsImVic2NvX3B1YV9pZCI6IjA3MjQyZjlhIn0.nRe7fIjMAIf1SkpFOTAEga3Gxx_Tk_QUgwATrgZpGfg"
```
Comment (esimons) [inline examples.json]: No `default`, right?
```suggestion
```

File: examples.json [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": { "$ref": "#/definitions/BookmarkRecord" }
+          },
+          "400": { "description": "Invalid ID supplied" },
```
Comment (esimons) [inline examples.json]: If a bad ID is supplied, should be a 404. 400 could be useful for indicating that an expected ID token wasn't provided though.

File: examples.json [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+          "404": { "description": "Bookmark not found" }
```
Comment (esimons) [inline examples.json]: Might want to add 401/403 as well.

File: examples.json [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+            "name": "bookmarkId",
+            "in": "path",
+            "description": "Bookmark id to delete",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "400": { "description": "Invalid ID supplied" },
+          "404": { "description": "Bookmark not found" }
```
Comment (esimons) [inline examples.json]: Might want to add 401/403 as well.

File: examples.json [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+          {
+            "name": "bookmarkId",
+            "in": "path",
+            "description": "Bookmark id to delete",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "400": { "description": "Invalid ID supplied" },
```
Comment (esimons) [inline examples.json]: If a bad ID is supplied, should be a 404. 400 could be useful for indicating that an expected ID token wasn't provided though.

File: examples.json [hunk 10]
```diff
@@ -0,0 +1,347 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": { "$ref": "#/definitions/BookmarkRecord" }
+          },
+          "400": { "description": "Invalid ID supplied" },
+          "404": { "description": "Bookmark not found" }
+        },
+        "security": [{ "api_key": [] }]
```
Comment (esimons) [inline examples.json]:
```suggestion
        "security": [{ "bookmarks_auth": ["read:bookmarks"] }]
```

File: examples.json [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+                  "$ref": "#/definitions/BookmarkItem"
+                },
+                "tags": {
+                  "$ref": "#/definitions/Tags"
+                }
+              }
+            }
+          }
+        ],
+        "responses": { "405": { "description": "Invalid input" } },
```
Comment (esimons) [inline examples.json]:
```suggestion
        "responses": { "400": { "description": "Invalid input" } },
```

File: examples.json [hunk 12]
```diff
@@ -0,0 +1,347 @@
...
+                      "href": "https://api.ebsco.com/v1/search/bookmarks?page_size=10&page=5"
+                    }
+                  ]
+                }
+              }
+            }
+          },
+          "404": { "description": "Bookmarks not found" }
+        },
+        "security": [{ "api_key": [] }]
```
Comment (esimons) [inline examples.json]:
```suggestion
        "security": [{ "bookmarks_auth": ["read:bookmarks"] }]
```

File: examples.json [hunk 13]
```diff
@@ -0,0 +1,347 @@
...
+                    {
+                      "rel": "last",
+                      "href": "https://api.ebsco.com/v1/search/bookmarks?page_size=10&page=5"
+                    }
+                  ]
+                }
+              }
+            }
+          },
+          "404": { "description": "Bookmarks not found" }
```
Comment (esimons) [inline examples.json]: Don't think this could ever return a 404, right?
```suggestion
```

File: examples.json [hunk 14]
```diff
@@ -0,0 +1,347 @@
...
+            "type": "integer"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "bookmarks": {
```
Comment (esimons) [inline examples.json]: Just a note -- if we want to standardize on `items` per the apparent v2 Paypal API spec, we'd want to change this here, too.
```suggestion
                "items": {
```

File: examples.json [hunk 15]
```diff
@@ -0,0 +1,347 @@
...
+        "security": [{ "api_key": [] }]
+      },
+      "post": {
+        "tags": ["bookmarks"],
+        "summary": "Add a new bookmark to the store",
+        "description": "",
+        "operationId": "addBookmark",
+        "consumes": ["application/json", "application/xml"],
+        "produces": ["application/xml", "application/json"],
+        "parameters": [
```
Comment (esimons) [inline examples.json]: We'd want to add an idempotency key to this example (once we figure out what that looks like).

File: examples.json (hunk 11 above)
Comment (esimons) [inline examples.json]: Should add a `201` as a possible response.

File: examples.json (hunk 9 above)
Comment (esimons) [inline examples.json]: Should add a `204` as a possible response.

File: examples.json [hunk 16]
```diff
@@ -0,0 +1,347 @@
...
+                "total_items": {
+                  "type": "integer",
+                  "example": 10
+                },
+                "total_pages": {
+                  "type": "integer",
+                  "example": 6
+                },
+                "links": {
+                  "type": "array",
```
Comment (esimons) [inline examples.json]: Imagine we'd want to provide a schema for link items as well... Woof, though.

File: examples.json (hunk 14 above)
Comment (ianqueue) [reply examples.json]: fair enough... I think they may have it wrong in their own patterns.md file them... https://github.com/paypal/api-standards/blob/master/patterns.md#page-navigation notice plans

File: examples.json (hunk 9 above)
Comment (ianqueue) [reply examples.json]: done

File: examples.json (hunk 11 above)
Comment (ianqueue) [reply examples.json]: done

File: examples.json (hunk 15 above)
Comment (ianqueue) [reply examples.json]: k

File: examples.json (hunk 14 above)
Comment (ianqueue) [reply examples.json]: done

File: examples.json (hunk 5 above)
Comment (ianqueue) [reply examples.json]: fixored

File: examples.json (hunk 4 above)
Comment (ianqueue) [reply examples.json]: yep

File: examples.json (hunk 3 above)
Comment (ianqueue) [reply examples.json]: me neither, but looked fun... removing for now

File: examples.json (hunk 2 above)
Comment (ianqueue) [reply examples.json]: done

File: examples.json (hunk 14 above)
Comment (esimons) [reply examples.json]: Yeah they absolutely have their wires crossed in their own docs (first noticed that in a PR I opened up over here, if you wanna take a look: https://github.com/EBSCOIS/api-standards/pull/12)

**Comment by esimons** [comment]
@ianqueue I think there were a few comments missed because Github collapsed them for whatever reason.

File: examples.json [hunk 17]
```diff
@@ -0,0 +1,416 @@
...
+          {
+            "name": "bookmarkId",
+            "in": "path",
+            "description": "Bookmark id to delete",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (esimons) [inline examples.json]: Should be a 204 according to the README: https://github.com/EBSCOIS/api-standards/blob/master/README.md#http-method-to-status-code-mapping

Comment (ianqueue) [reply examples.json]: done

**Comment by esimons** [comment]
Also I declare myself tapped out on making comments like "*Actually*, this ought to be a 204 not a 200". Gonna need to wash the bad taste out of my mouth after this.

**Pull request merged**